x-pack/plugins/cases/server/services/user_actions/operations/create.ts

```diff
@@ -154,6 +162,52 @@ export class UserActionPersister {
     return this.buildAddDeleteUserActions(params, createPayload, UserActionTypes.tags);
   }
 
+  private buildCustomFieldsUserActions(params: TypedUserActionDiffedItems<CaseCustomField>) {
+    const createPayload: CreatePayloadFunction<
+      CaseCustomField,
+      typeof UserActionTypes.customFields
+    > = (items: CaseCustomFields) => ({ customFields: items });
+
+    const { originalValue, newValue } = params;
+    const compareValues = arraysDifference(originalValue, newValue);
+    const updatedCustomFieldsKeys = new Set(
+      compareValues?.addedItems.map((customField) => customField.key) ?? []
+    );
+
+    const updatedCustomFieldsUsersActions = compareValues?.addedItems
+      .map((customField) =>
+        this.buildUserAction({
+          commonArgs: params,
+          actionType: UserActionTypes.customFields,
+          action: UserActionActions.update,
+          createPayload,
+          modifiedItems: [customField],
+        })
+      )
+      .filter((userAction): userAction is UserActionEvent => userAction != null);
+
+    const customFieldsRemovedUserActions = compareValues?.deletedItems
+      .map((customField) => {
+        if (updatedCustomFieldsKeys.has(customField.key)) {
```

Comment:
The `arraysDifference` will return the old value as a deleted item. We do not create user actions for the old values as they are covered as updated user actions.